p8e-util/src/main/kotlin/io/p8e/crypto/SignerFactory.kt

```diff
@@ -0,0 +1,20 @@
+package io.p8e.crypto
+
+import java.lang.IllegalStateException
+import java.security.KeyPair
+
+class SignerFactory(
+    smartKeyApiKey: String
+){
+    private val appApiKey = smartKeyApiKey
+
+    fun getSigner(uuid: String? = null, keyPair: KeyPair? = null): SignerImpl {
```

Comment:
this could be more kotlin idiomatic with a sealed class. For instance the idea would look like this.

```kotlin
sealed class Signer
data class KeyPairSigner(val keyPair: KeyPair) : Signer
data class SmartKeySigner(val uuid: String) : Signer

fun getSigner(signer: Signer): SignerImpl {
    return when (signer) {
        is KeyPairSigner -> Pen(signer.keyPair)
        is SmartKeySigner -> SmartKeySigner(appApiKey, signer.uuid)
    }
}
```

Reply:
i had something similar last night, in my mind i thought was this was like explicitly telling which signer to use, ill put up what i had back.